### PR TITLE
Clustering added.

### PR DESCRIPTION
**Pull request by bsircelj**
Added a method to compute clusters from selected features in EOPatch.

### CODE REVIEW

File: features/eolearn/features/clustering.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        :param eopatch: EOPatch with all features that will be used
+        :param features: Features for clustering
+        :return: array of vectors constructed from the features listed
+        """
+
+        feature = EOTask._parse_features(features)
+        data = None
+        for i in feature:
+            if data is None:
+                data = eopatch.data_timeless[i[1]]
```
Comment (azupanc) [inline features/eolearn/features/clustering.py]: Tha tasks seems to assume that clusters will be always determined using `data_timeless` feature. If this is the case, than make this explicit in the task constructor:

```
self.features  = self._parse_features(features, default_feature_type=FeatureType.DATA_TIMELESS)
```

File: features/eolearn/features/clustering.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+
+class ClusteringTask(EOTask):
+    """
+    Tasks computes clusters on selected features using sklearn.cluster.AgglomerativeClustering.
+
+    The algorithm produces a timeless data feature where each cell has a natural number which corresponds to specific
+    group. The cells marked with -1 are not marking clusters. They are either being excluded by a mask or later removed
+    by depending on the 'remove_small' threshold.
+    """
```
Comment (azupanc) [inline features/eolearn/features/clustering.py]: Explicitly mention in the documentation that input features need to be timeless.

**Comment by mlubej** [comment]
@azupanc did you have any other comments for this PR or is everything solved?

If yes, I would suggest @bsircelj to merge the latest develop into his branch and fix the failing checks.

**Comment by mlubej** [comment]
>  Merge pull request #9 from sentinel-hub/master 

probably you meant from develop, not master?

**Comment by mlubej** [comment]
Thanks!

**Pull request merged**